miner/worker.go

```diff
@@ -878,6 +878,17 @@ func (w *worker) commitNewTx(tx *types.Transaction) error {
 	timestamp := tx.L1Timestamp()
 
 	num := parent.Number()
+	// Fill in the index field in the tx meta if it is `nil`.
+	// This should only ever happen in the case of the sequencer
+	// receiving a queue origin sequencer transaction. The verifier
+	// should always receive transactions with an index as they
+	// have already been confirmed in the canonical transaction chain.
+	// Use the parent's block number because the CTC is 0 indexed.
+	if meta := tx.GetMeta(); meta.Index == nil {
```

Comment:
A bit confused about when this would happen. Could you elaborate a little more?

Reply:
A transaction will definitely have an index when it has been confirmed in the CTC. Before being confirmed in the CTC, there is no concept of an index. This code path should only be hit when it is a queue origin sequencer transaction sent to the actual sequencer. Verifiers will always receive transactions with an index, since by definition the tx has been included in the CTC. The sequencer adds the index here right before it confirms the transaction so that the replicas (dtl sequencer sync) can have an index to work with

Reply:
Ahhh okay perfect, I see. Looks good to me!